src/sort.rs

```diff
@@ -68,7 +90,27 @@ pub fn sort_toml(
     let mut first_table = None;
     let mut heading_order: BTreeMap<_, Vec<Heading>> = BTreeMap::new();
     for (idx, (head, item)) in toml.as_table_mut().iter_mut().enumerate() {
-        if !matcher.heading.contains(&head.get()) {
+        let mut target_tables = BTreeMap::new();
+        let item_key = head.get();
+        if item_key == TARGET {
+            if let Some(table) = item.as_table() {
+                for key in matcher.heading {
```

Comment:
By comparing against all headings, we theoretically would also sort tables like `target.cfg.package` right? Those don't make any sense so perhaps we can hardcode the 3 types of dependencies here?